statuspage_connector.py

```diff
@@ -0,0 +1,591 @@
+
+        data['incident'] = incident
+
+        endpoint = 'pages/{0}/incidents/{1}'.format(self._page_id, incident_id)  # pragma: allowlist secret
+        ret_val, response = self._make_rest_call(endpoint, action_result, json=data, method="patch")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully updated incident")
+
+    def _get_fields(self, param, action_result):
+
+        components = param.get('components')
+
+        # fields is an optional field
+        if not components:
+            return RetVal(phantom.APP_SUCCESS, None)
+
+        # we take in as a dictionary string, first try to load it as is
+        try:
+            components = json.loads(components)
+        except Exception as e:
+            error_msg = self._get_error_message_from_exception(e)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, STATUSPAGE_ERR_FIELDS_JSON_PARSE.format(error=error_msg)), None)
+
+        return RetVal(phantom.APP_SUCCESS, components)
+
+    def _handle_create_incident_subscriber(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        incident_id = param['incident_id']
+
+        data = dict()
+        data['subscriber'] = {
+            'email': param['email']
+            # 'phone_country': param.get('phone_country'),
+            # 'phone_number': param.get('phone_number'),
+            # 'skip_confirmation_notification': param.get('skip_confirmation_notification')
+        }
+
+        endpoint = 'pages/{0}/incidents/{1}/subscribers'.format(self._page_id, incident_id)  # pragma: allowlist secret
+        ret_val, response = self._make_rest_call(endpoint, action_result, json=data, method='post')
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully created incident subscriber")
+
+    def _handle_list_incident_subscribers(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        incident_id = param.get('incident_id')
+        endpoint = 'pages/{0}/incidents/{1}/subscribers'.format(self._page_id, incident_id)  # pragma: allowlist secret
+
+        ret_val, response = self._make_rest_call(endpoint, action_result)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        for incident in response:
+            action_result.add_data(incident)
+
+        summary = action_result.set_summary({})
+        summary['num of subscribers'] = len(response)
```

Comment:
Remove spaces from the Summary key and respective data_paths and use usderscore.

Reply:
Done